modules/gdscript/gdscript_parser.cpp

```diff
@@ -5919,7 +5919,7 @@ GDScriptParser::DataType GDScriptParser::_get_operation_type(const Variant::Oper
 	if (b_type == Variant::INT || b_type == Variant::REAL) {
 		Variant::evaluate(Variant::OP_ADD, b, 1, b, r_valid);
 	}
-	if (a_type == Variant::STRING && b_type != Variant::ARRAY) {
+	if (a_type == Variant::STRING && b_type != Variant::ARRAY && b_type != Variant::POOL_STRING_ARRAY && b_type != Variant::POOL_INT_ARRAY && b_type != Variant::POOL_BYTE_ARRAY && b_type != Variant::POOL_COLOR_ARRAY && b_type != Variant::POOL_REAL_ARRAY && b_type != Variant::POOL_VECTOR2_ARRAY && b_type != Variant::POOL_VECTOR3_ARRAY) {
```

Comment:
Is there another way of doing this? it feels bad to write that number of conditions. I know they are needed in order to pass the operation evaluation of the script parsing.

Reply:
ok, I've been checking the variant.h file and I've seen the `is_array()` method, I'll use it instead.